modules/aql/arrow-aql/src/main/kotlin/arrow/aql/Max.kt

```diff
@@ -4,20 +4,25 @@ import arrow.core.ForId
 import arrow.core.Id
 import arrow.core.identity
 import arrow.core.value
+import arrow.core.None
+import arrow.core.Some
+import arrow.core.Option
 import arrow.typeclasses.Foldable
-import kotlin.math.max
+import arrow.typeclasses.Order
 
 interface Max<F> {
 
   fun foldable(): Foldable<F>
 
-  infix fun <A, Z> Query<F, A, Z>.max(f: A.() -> Long): Query<ForId, Long, Long> =
-    foldable().run {
-      Query(
-        select = ::identity,
-        from = Id(from.foldLeft(Long.MIN_VALUE) { acc, a -> max(f(a), acc) })
-      )
-    }
+  fun <A, X, Z> Query<F, A, Z>.max(ord: Order<X>, f: A.() -> X): Query<ForId, Option<X>, Option<X>> =
+    Query(
+      select = ::identity,
+      from = Id(foldable().run {
+        from.foldLeft(None) { acc: Option<X>, a: A ->
+          acc.fold({ Some(f(a)) },
+            { Some(if (ord.run { it > (f(a)) }) it else f(a)) })
```

Comment:
Nit: `(f(a))` doesn't need the additional parenthesis.

Reply:
ups, fixed!